code/week7/generic_class.h

```diff
@@ -0,0 +1,16 @@
+#include <iostream>
+
+template <class primitive_type> class Example {
+private:
+    primitive_type var;
+public:
+    Example(primitive_type var) : var(var) {
+        std::cout << "creating object with var = " << var << std::endl;
+    };
+    ~Example() {};
+
+    primitive_type getVar() {
```

Comment:
I think it's best practice to mark this function as const, but I don't know if we want to make it easy to read or not at this point.